modules/EvseManager/IECStateMachine.cpp

```diff
@@ -505,17 +505,26 @@ void IECStateMachine::connector_force_unlock() {
         cp = cp_state;
     }
 
+    if (not relais_on) {
+        // Unconditionally try to unlock, as `is_locked` might not always reflect the physical state of the lock.
+        // This can occur for example in case of a failed unlock due to a hardware issue.
+        signal_unlock();
+        is_locked = false;
+    }
+
     if (cp == RawCPState::B or cp == RawCPState::C) {
         force_unlocked = true;
         check_connector_lock();
     }
 }
 
 void IECStateMachine::check_connector_lock() {
-    if (should_be_locked and not force_unlocked and not is_locked) {
```

Comment:
Previously, relais_on condition was not here. This meant that if force_unlock was on in cp state B, and then the state changed to C, the relais will not lock.